software/mk/Makefile.builddefs

```diff
@@ -123,24 +124,29 @@ CRT_OBJ ?= crt.o
 
 ifeq ($(BSG_ELF_OFF_CHIP_MEM), 1)
   ifeq ($(BSG_ELF_DEFAULT_DATA_LOC), LOCAL)
-    LINK_SCRIPT ?= $(BSG_MANYCORE_DIR)/software/spmd/common/link_dmem.ld
+    LINK_GEN_OPTS ?= --default_data_loc=dmem --dram_size=$(BSG_ELF_DRAM_SIZE) --sp=$(BSG_ELF_STACK_PTR)
   else ifeq ($(BSG_ELF_DEFAULT_DATA_LOC), SHARED)
-    LINK_SCRIPT ?= $(BSG_MANYCORE_DIR)/software/spmd/common/link_dram.ld
+    LINK_GEN_OPTS ?= --default_data_loc=dram --dram_size=$(BSG_ELF_DRAM_SIZE) --sp=$(BSG_ELF_STACK_PTR)
   else
     $(error Invalid BSG_ELF_DEFAULT_DATA_LOC = $(BSG_ELF_DEFAULT_DATA_LOC); Only LOCAL and SHARED are valid)
   endif
 else ifeq ($(BSG_ELF_OFF_CHIP_MEM), 0)
   ifeq ($(BSG_ELF_DEFAULT_DATA_LOC), LOCAL)
-    LINK_SCRIPT ?= $(BSG_MANYCORE_DIR)/software/spmd/common/link_dmem2.ld
+    LINK_GEN_OPTS ?= --default_data_loc=dmem --dram_size=$(BSG_ELF_VCACHE_SIZE) --sp=$(BSG_ELF_STACK_PTR)
   else ifeq ($(BSG_ELF_DEFAULT_DATA_LOC), SHARED)
-    LINK_SCRIPT ?= $(BSG_MANYCORE_DIR)/software/spmd/common/link_dram2.ld
+    LINK_GEN_OPTS ?= --default_data_loc=dram --dram_size=$(BSG_ELF_VCACHE_SIZE) --sp=$(BSG_ELF_STACK_PTR)
```

Comment:
A general question for discussion: 

Effectively, these options are determined by the machine file. Should we put them there, so that users can more easily change them?

Reply:
Some of then differ program to program and some of then are taken from machine file anyway...

Reply:
The script doesn't have any global parameters. Everything configurable, is provided by command line arguments.